File: Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) == 1:
            return intervals

        intervals = sorted(intervals, key=lambda x: x[0])
        new_intervals = []

        i = 0

        while (i < len(intervals)):
            s, e = intervals[i]
            i += 1
            while (i < len(intervals)):
                s_c, e_c = intervals[i]
                if e >= s_c:
                    e = max(e_c, e)
                    s_c, e_c = intervals[i]
                    i += 1
                else:
                    break

            new_intervals.append([s, e])

        return new_intervals

File: java_version/src/mergeIntervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> Integer.compare(a[0], b[0]));
        ArrayList<int[]> inter = new ArrayList<int[]>();
        inter.add(intervals[0]);
        int x = 0;
        for (int i = 1; i < intervals.length; i++){
            if (intervals[i][0] <= inter.get(x)[1]){
                if (intervals[i][1] > inter.get(x)[1])
                    inter.get(x)[1] = intervals[i][1];
            }
            else{
                inter.add(intervals[i]);
                x++;
            }
                
        }
        int[][] res = new int[inter.size()][2];
        for (int i=0; i< inter.size(); i++){
            res[i] = inter.get(i);
        }
        return res;
        
    }
}